core/templates/base-components/oppia-root.directive.ts

```diff
@@ -0,0 +1,84 @@
+/* eslint-disable max-len */
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview The root directive of the oppia ajs application.
+ */
+
+// Incase of doubts over what is done here, please look at the description of
+// pr #9479
+
+import { OppiaAngularRootComponent } from
+  'components/oppia-angular-root.component';
+angular.module('oppia').directive('oppiaRoot', [
+  function() {
+    return {
+      template: require('./oppia-root.directive.html'),
+      scope: {},
+      transclude: {
+        app: 'app',
+      },
+      controllerAs: '$ctrl',
+      controller: ['$scope',
+        function($scope) {
+          $scope.initialized = false;
+
+          $scope.onInit = function() {
+            // This config is not getting executed. This doesn't affect my
```

Comment:
I have kept it for now. I really want this DI to happen so that we can get rid of most of UpgradedServices. I said most of UpgradedServices because I think we may still need to provide some utility services like UrlInterpolationService, etc. They are used in header and footer and that loads before the angular app loads (otherwise users would be looking at a white screen. We wouldn't want to have this, even if it is for 80ms).